package.json

```diff
@@ -0,0 +1,39 @@
+{
+  "name": "openLRSng-configurator",
+  "description": "openLRSng Configurator",
+  "version": "3.8.8",
+  "main": "main.html",
+  "default_locale": "en",
+  "scripts": {
+    "start": "node_modules/gulp/bin/gulp.js build && node_modules/nw/bin/nw ."
```

Comment:
This fails for me:

```
user@pc:~/git/openLRSng-configurator$ npm start

> openLRSng-configurator@3.8.8 start /home/user/git/openLRSng-configurator
> node_modules/gulp/bin/gulp.js build && node_modules/nw/bin/nw .

[13:34:27] No gulpfile found
```

Can you please fix?

Reply:
I first included the package.json file in the openlrsng repository files. Then y created a zip file manually selecting all files (and folders), and rename it to openLRSng-configurator.nw.

After dat I went to the nw.js respository and downloaded the were I had decompressed the nwjs zip file for win32: [https://github.com/nwjs/nw.js#downloads](https://github.com/nwjs/nw.js#downloads), and unziped it.

Then I moved the openLRSng-configurator.nw file into the nwsjs folder and executed this command from cmd:

copy /b nw.exe+openLRSng-configurator.nw openLRSng-configurator.exe

After that I removed files nw.exe and openLRSng-configurator.nw, then I zipped everything toguether and uploaded it to the releases folder of the repository.

Thas all, and is exactly the same that betaflight or iNav standalone apps:

![betaflight-vs-openlrsng-standalone](https://user-images.githubusercontent.com/12935392/34415625-138a7036-ebf0-11e7-98e6-026b061118c9.png)

Reply:
I can't quite see how this description relates to the fact that the `start` script in your script seems to be broken. If it's meant to work it should be fixed, if it's not meant to work it should not be in there in the first place.

Also I think that having to do these steps manually will lead to poor repeatability for building / testing, and should be avoided.

Have a look at https://github.com/betaflight/betaflight-configurator for an example of how the build can be completely automated `npm start` will actually build, and then start the configurator in Betaflight.

Reply:
I agree with you about the need to automate the building process for developing. If I have more time I'll help you doing it. By now I only wanted to make a litle contribution to help end users to run the configurator without the need of Google Chrome. I thik you could build it manually for the last version and publish it, so future users could download and run it.

Reply:
Thank you for trying to help.

At this point I don't see much value in adding an essentially empty `package.json` (once you've removed the things that are broken or unneeded), and then expecting users to manually install all of the parts of the toolchain and do the build.
If they want to do the build manually, they can just as well execute a `npm init` as the first step, in order to create an empty `package.json`.

Reply:
I don't want end users to build it. What I mean is to suggesst to you to build it manually and publish it under releases section for future users, until the developers automate it.

Reply:
You are suggesting we manually install and then manually build / pack the app on 3 platforms that should be supported (linux, macos, windows). And you hope that we will get it right and manage to produce 3 identical builds. And then we do this again all over, if a bug is reported and fixed?

This will require a lot of time (and access to 3 different build machines). Time that could otherwise be spent on implementing an automated build.

Another consideration is that the app in it's current state is not ready to work as a standalone app: It relies on one firmware version being packaged inside the app, and the app only being able to install this one version, or a custom built, not verified binary. This was a good approach while the app was delivered through the Chrome web store, since it essentially leveraged the web store to push the firmware onto the users' workstations. For the standalone app this won't work - we can't expect users to download and install a new release of the > 80 MB standalone application just to get a new firmware release. Have a look at Betaflight configurator to see how it is capable of getting the latest firmware release over the internet, in a user friendly way. This is what we should be going for in the standalone app.

So to sum it up, I don't see why we should be offering users an incomplete and unreliable standalone app - they will be better off if they continue to use the Chrome web store app until the standalone app (including the build process) is finalised and ready for release.

Reply:
And when do you spect to be released???

Reply:
When it's ready for release - within the next month or so.

Reply:
Ok I see reasonable to wait for then.